rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRest.java

```diff
@@ -1066,4 +1106,56 @@ public Response index() throws URISyntaxException {
         return Response.seeOther(new URI("doc/jaxrsdocs/rm/index.html")).build();
     }
 
+    private void prepareInfraParams(@FormParam("infrastructureType") String infrastructureType,
+            @FormParam("infrastructureParameters") String[] infrastructureParameters,
+            @FormParam("infrastructureFileParameters") String[] infrastructureFileParameters, ResourceManager rm,
+            Object[] infraParams) {
+        /*
+         * we need to merge both infrastructureParameters and infrastructureFileParameters into one
+         * to do so we need the infrastructure parameter order from the RM
+         */
+        for (PluginDescriptor infra : rm.getSupportedNodeSourceInfrastructures()) {
+            if (infra.getPluginName().equals(infrastructureType)) {
+                int i = 0, j = 0, k = 0;
+                for (ConfigurableField field : infra.getConfigurableFields()) {
+                    if (field.getMeta().credential() || field.getMeta().fileBrowser()) {
+                        // file parameter : insert from the other array, convert
+                        // to byte[]
+                        infraParams[i] = infrastructureFileParameters[k].getBytes();
+                        k++;
+                    } else {
+                        infraParams[i] = infrastructureParameters[j];
+                        j++;
+                    }
+                    i++;
+                }
+            }
+        }
+    }
+
+    private void preparePolicyParams(@FormParam("policyType") String policyType,
+            @FormParam("policyParameters") String[] policyParameters,
+            @FormParam("policyFileParameters") String[] policyFileParameters, ResourceManager rm,
+            Object[] policyParams) {
+        for (PluginDescriptor pol : rm.getSupportedNodeSourcePolicies()) {
```

Comment:
Can you refactor this in favor for less nested loops and conditionals

Reply:
done